pkg/probe/probes.go

```diff
@@ -77,174 +77,165 @@ func currentStateAsGJson() (gjson.Result, error) {
-		currentStateAsGJson, err := currentStateAsGJson()
-		if err != nil {
-			return false, errors.Wrap(err, "failed retrieving current state to get name resolving config")
-		}
-		runningNameServers = currentStateAsGJson.Get(runningServersGJsonPath).Array()
-		if len(runningNameServers) == 0 {
-			return false, fmt.Errorf("missing name servers at '%s' on %s", runningServersGJsonPath, currentStateAsGJson.String())
+	// Get the name servers at node since the ones at container may not be up to date
+	currentStateAsGJson, err := currentStateAsGJson()
+	if err != nil {
+		return errors.Wrap(err, "failed retrieving current state to get name resolving config")
+	}
+	runningNameServers := currentStateAsGJson.Get(runningServersGJsonPath).Array()
+	if len(runningNameServers) == 0 {
+		return fmt.Errorf("missing name servers at '%s' on %s", runningServersGJsonPath, currentStateAsGJson.String())
+	}
+	for _, runningNameServer := range runningNameServers {
+		r := &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{
+					Timeout: defaultDNSProbeTimeout,
+				}
+				return d.DialContext(ctx, network, net.JoinHostPort(runningNameServer.String(), "53"))
+			},
 		}
-		for _, runningNameServer := range runningNameServers {
-			r := &net.Resolver{
-				PreferGo: true,
-				Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
-					d := net.Dialer{
-						Timeout: timeout,
-					}
-					return d.DialContext(ctx, network, net.JoinHostPort(runningNameServer.String(), "53"))
-				},
-			}
-			_, err := r.LookupNS(context.TODO(), "root-server.net")
-			if err != nil {
-				errs = append(errs, err)
-			} else {
-				return true, nil
-			}
+		_, err := r.LookupNS(context.TODO(), "root-server.net")
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			return nil
 		}
-		return false, fmt.Errorf("failed checking DNS connectivity: %v", errs)
-	})
+	}
+	return fmt.Errorf("failed checking DNS connectivity: %v", errs)
 }
 
 // Select will return the external connectivity probes that are working (ping and dns) and
 // the internal connectivity probes
 func Select(cli client.Client) []Probe {
 	probes := []Probe{}
-
-	err := runPing(cli, time.Second)
-	if err == nil {
-		probes = append(probes, Probe{
+	externalConnectivityProbes := []Probe{
+		{
 			name:    "ping",
 			timeout: defaultGwProbeTimeout,
 			run:     runPing,
-		})
-	} else {
-		log.Info("WARNING not selecting 'ping' probe")
-	}
-	err = runDNS(cli, time.Second)
-	if err == nil {
-		probes = append(probes, Probe{
+		},
+		{
 			name:    "dns",
 			timeout: defaultDNSProbeTimeout,
 			run:     runDNS,
+		},
+	}
+
+	for _, p := range externalConnectivityProbes {
+		err := wait.PollImmediate(time.Second, p.timeout, func() (bool, error) {
+			err := p.run(cli)
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
```

Comment:
Let's create a "factory function" with cli that returns `wait.ConditionFunc` function expected by `PollImmediate`, this will simplify the polling code 

```golang
func pingCondition(_ client.Client) wait.ConditionFunc {
    return func() (bool, error) {
       ...
       return ping(defaultGw)
    }
}


func dnsCondition(_ client.Client) wait.ConditionFunc {
   return func() (bool, error) {
   ....
   }
}

{
			name:    "dns",
			timeout: defaultDNSProbeTimeout,
			condition:     dnsCondition,
		},

err := wait.PollImmediate(time.Second, p.timeout, p.condition(cli))
if err != nil {
...
}
`
```

Reply:
Done